src/environment.cpp

```diff
@@ -218,7 +218,16 @@ ABMWithState::ABMWithState(ActiveBlockModifier *abm_, ServerEnvironment *senv):
 {
 	auto ndef = senv->getGameDef()->ndef();
 	interval = abm->getTriggerInterval();
+	if (!interval)
+		interval = 10;
 	chance = abm->getTriggerChance();
+	if (!chance)
+		chance = 10;
+
+	neighbors_range = abm->getNeighborsRange();
+	if (!neighbors_range)
+		neighbors_range = 1;
```

Comment:
shouldn't default values be handled in abm class? i.e. LuaABM since it's the only one that may return zero values. Also why are there defaults, shouldn't we just fail at register_abm step instead?